### ISSUE TITLE
full syntax support

### ISSUE THREAD

**Issue by KutnerUri** [snyamathi/semver-intersect#14]
Great work with this package, I am trying to use it in my project.

Sadly I am getting exceptions on the **|| operator**, even though it's valid syntax.
(eg. ```"react": "~0.14.0 || ^15.0.0 || ^16.0.0"```)

Seems like it is the only operator missing, as all the others seem to be supported:
- version range (<, >, <=, >=, =)
- v prefix (v15.2.3)
- tags (15.2.3-tag.0)
- hyphen (15.2.3 - 15.4.2)
- tilde (~15.3.2)
- caret ( ^15.2.3)
- wildcard ('*', 'x', or 'X')

**Comment by snyamathi** [comment]
I do recall trying to get this to work with multiple `||` separated ranges, though I don't remember why it wasn't implemented.  I can take a look at this, in the near future but no ETA specifically unless you want to look into a PR.  The code is a little crazy, but the test suite should keep you from breaking anything.

I think one thing to spec out would be the expectation of how these ranges would be handled, with a few real world examples.

**Comment by KutnerUri** [comment]
thanks for the quick reply.

I am trying to install nested peer dependencies, so I am getting things like
```
["1 || 2 || ^2.1.0-beta || ^2.2.0-rc", "^1.0.0 || ^2.0.0 || ^3.0.0"]
["^15.6.2", "^15.6.2", "^0.14.0 || ^15.0.0 || ^16.0.0-beta || ^16.0.0"]
```

**Comment by Silic0nS0ldier** [comment]
I've got a quick-and-dirty workaround for the `||` syntax in https://github.com/userfrosting/merge-package-dependencies, which does the trick but is verbose compared to what this package produces. It boils down to splitting on `||`, combining each piece with the next range. Lots of room for improvement (e.g. eliminating combinations that lack overlap).

**Comment by nicky1038** [comment]
Hi everyone!

I've added support for logical-or operators in the following [PR](https://github.com/snyamathi/semver-intersect/pull/19).

To make this happen, I had to additionally split range strings by "||" and to work with nested arrays, making the logic of `reduce` callback in the main function a little bit more complicated. Though not much was changed.

 There is also support for versions "*", "x" and "X" in the same PR.
